### DISCUSSION TITLE
How to improve session creation latency

### DISCUSSION THREAD

**Discussion by ashep** [zitadel/zitadel#8286] [Q&A] [answered]
Hello there!

On my setup, it takes Zitadel about 5-6 seconds to process a user [session creation](https://zitadel.com/docs/apis/resources/session_service/session-service-create-session) request, which is quite a long time.

Could you please suggest any approach to reduce this time as much as possible?

My setup is:

- AWS ECS instance (0.25 CPU, 512 MB RAM)
- AWS RDS [db.t4g.small](https://aws.amazon.com/rds/instance-types/) (2GB RAM, 2 vCPU), PostgreSQL 16.2.

Thank you.

**Comment by ashep** [comment, marked as answer]
After increasing ECS instance to 1 vCPU/2GB RAM, the response time decreased to 1 second.

**Reply by hifabienne** [reply]
Thanks for your feedback
